packages/lib/services/plugins/RepositoryApi.ts

```diff
@@ -165,24 +191,62 @@ export default class RepositoryApi {
 		}
 	}
 
-	public async search(query: string): Promise<PluginManifest[]> {
-		query = query.toLowerCase().trim();
+	public async searchAndFilter(category?: string, searchQuery?: string): Promise<PluginManifest[]> {
+		const manifests: PluginManifest[] = await this.manifests();
+		let output: PluginManifest[] = [];
+		const pluginStates = Setting.value('plugins.states');
+		category = category.toLowerCase();
 
-		const manifests = await this.manifests();
-		const output: PluginManifest[] = [];
-
-		for (const manifest of manifests) {
-			for (const field of ['name', 'description']) {
-				const v = (manifest as any)[field];
-				if (!v) continue;
+		const categoryFilterList = ['appearance', 'developer tools', 'productivity', 'themes', 'integrations', 'viewer', 'search', 'tags', 'editor', 'files', 'personal knowledge management'];
 
-				if (v.toLowerCase().indexOf(query) >= 0) {
-					output.push(manifest);
-					break;
-				}
+		if (categoryFilterList.includes(category)) {
+			output = manifests.filter((m) => m.categories && m.categories.includes(category));
+		} else {
+			switch (category) {
+			case 'most downloaded':
+				output = manifests.sort((m1, m2) => m2._totalDownloads - m1._totalDownloads).slice(0, 50);
+				break;
+			case 'recommended':
+				output = manifests.filter(manifest => manifest._recommended);
+				break;
+			case 'newest':
+				// + before new keyword forces date to be number https://github.com/microsoft/TypeScript/issues/5710
+				output = manifests.sort((m1, m2) => +new Date(m2._created_date) - +new Date(m1._created_date)).slice(0,50);
```

Comment:
How do you define "newest"? Last updated? Last created? Do we actually need this category?